src/compute-client/src/controller.rs

```diff
@@ -489,6 +533,106 @@ where
+    pub async fn create_dataflows(
+        &mut self,
+        instance_id: ComputeInstanceId,
+        dataflows: Vec<DataflowDescription<crate::plan::Plan<T>, (), T>>,
+    ) -> Result<(), ComputeError> {
+        self.instance(instance_id)?
+            .create_dataflows(dataflows)
+            .await
+    }
+
+    /// Drop the read capability for the given collections and allow their resources to be
+    /// reclaimed.
+    pub async fn drop_collections(
+        &mut self,
+        instance_id: ComputeInstanceId,
+        collection_ids: Vec<GlobalId>,
+    ) -> Result<(), ComputeError> {
+        self.instance(instance_id)?
+            .drop_collections(collection_ids)
+            .await
+    }
+
+    /// Initiate a peek request for the contents of the given collection at `timestamp`.
+    pub async fn peek(
+        &mut self,
+        instance_id: ComputeInstanceId,
+        collection_id: GlobalId,
+        literal_constraints: Option<Vec<Row>>,
+        uuid: Uuid,
+        timestamp: T,
+        finishing: RowSetFinishing,
+        map_filter_project: mz_expr::SafeMfpPlan,
+        target_replica: Option<ReplicaId>,
+    ) -> Result<(), ComputeError> {
+        self.instance(instance_id)?
+            .peek(
+                collection_id,
+                literal_constraints,
+                uuid,
+                timestamp,
+                finishing,
+                map_filter_project,
+                target_replica,
+            )
+            .await
+    }
+
+    /// Cancel existing peek requests.
+    ///
+    /// Canceling a peek is best effort. The caller may see any of the following
+    /// after canceling a peek request:
+    ///
+    ///   * A `PeekResponse::Rows` indicating that the cancellation request did
+    ///    not take effect in time and the query succeeded.
+    ///
+    ///   * A `PeekResponse::Canceled` affirming that the peek was canceled.
+    ///
+    ///   * No `PeekResponse` at all.
+    pub async fn cancel_peeks(
+        &mut self,
+        instance_id: ComputeInstanceId,
+        uuids: &BTreeSet<Uuid>,
+    ) -> Result<(), ComputeError> {
+        self.instance(instance_id)?.cancel_peeks(uuids).await
+    }
+
+    /// Assign a read policy to specific identifiers.
+    ///
+    /// The policies are assigned in the order presented, and repeated identifiers should
+    /// conclude with the last policy. Changing a policy will immediately downgrade the read
+    /// capability if appropriate, but it will not "recover" the read capability if the prior
+    /// capability is already ahead of it.
+    ///
+    /// Identifiers not present in `policies` retain their existing read policies.
+    pub async fn set_read_policy(
+        &mut self,
+        instance_id: ComputeInstanceId,
+        policies: Vec<(GlobalId, ReadPolicy<T>)>,
+    ) -> Result<(), ComputeError> {
+        self.instance(instance_id)?.set_read_policy(policies).await
+    }
+
+    /// Update the max size in bytes of any result.
```

Comment:
(same here) Can you add in the docstring in which cases this method can fail?

I think it's in particular important to understand if the only failure mode is invalid id or if there are also other cases, that gives the caller some confidence whether to call `.unwrap()` or not

Btw, how does this actually work? What do we do if the result would exceed that size?